tools/swift-ide-test/ModuleAPIDiff.cpp

```diff
@@ -770,6 +770,7 @@ class SMAModelGenerator : public DeclVisitor<SMAModelGenerator> {
 
   std::vector<sma::TypeName> collectProtocolConformances(NominalTypeDecl *NTD) {
     std::vector<sma::TypeName> Result;
+    Result.reserve(NTD->getAllProtocols().size());
     for (const auto *PD : NTD->getAllProtocols()) {
```

Comment:
This is going to compute `getAllProtocols()` twice, so that's probably *worse* than whatever gains are saved here! Maybe a patch that stores to a temporary, if it's bothering you that much?

(This isn't a tool that runs in a time-sensitive manner in the first place, though.)

Reply:
That's fair. I will do the temp. Sorry I am super new to swift code base, could you guide me to where the most time-sensitive code might be? I would love to look at improving some performance critical code =)

Reply:
Done @jrose-apple =)